lib/remap-functions/src/redact.rs

```diff
@@ -108,9 +108,12 @@ impl Expression for RedactFn {
     }
 
     fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        use value::Kind;
+
         self.value
             .type_def(state)
-            .with_constraint(value::Kind::Bytes)
+            .fallible_unless(Kind::Bytes)
+            .with_constraint(Kind::Bytes)
```

Comment:
This type_def also needs to take into account the `patterns` parameter. This also raises an error because you are passing a parameter that is not a regex:

```
$ redact("noog oog", filters = ["pattern"], patterns = [2])
remap error: value error: expected string or regex, got "integer"
```

I think the whole thing should look more like:

```rust
        let mut typedef = self
            .value
            .type_def(state)
            .with_constraint(value::Kind::Bytes);

        match &self.patterns {
            Some(patterns) => {
                for p in patterns {
                    typedef = typedef.merge(p.type_def(state).with_constraint(value::Kind::Regex));
                }
            }
            None => (),
        }

        typedef
```

Reply:
So, what I think I'm still a bit confused about is why `patterns` needs to be accounted for here while `filters` and `redactor` don't. This, for example, would fail:

```js
redact("noog oog", filters = ["does not exist"], ...)
```

This would also fail:

```js
redact("noog oog", redactor = "oingo boingo", ...)
```